### PR TITLE
Upgrade Translate to latest Google API Client

### PR DESCRIPTION
**Pull request by quartzmo**
This PR completes Translate upgrade for PR GoogleCloudPlatform/gcloud-ruby#733.

`rubocop`, `rake test:translate`, and `rake test:acceptance:translate` all passing.

### CODE REVIEW

File: lib/gcloud/translate/api.rb [hunk 1]
```diff
@@ -196,9 +194,8 @@ def translate *text, to: nil, from: nil, format: nil, cid: nil
       def detect *text
         return nil if text.empty?
         text = Array(text).flatten
-        resp = connection.detect(text)
-        fail ApiError.from_response(resp) unless resp.success?
-        Detection.from_response resp, text
+        resp = service.detect(text)
```
Comment (blowmage) [inline lib/gcloud/translate/api.rb]: The variable is no longer a response object, but is the api object. Can you change the variable name to `gapi` to reflect this?

File: lib/gcloud/translate/translation.rb [hunk 2]
```diff
@@ -91,8 +91,8 @@ def detected?
       ##
       # @private New Translation from a TranslationsListResponse object as
       # defined by the Google API Client object.
-      def self.from_response resp, text, to, from
-        res = text.zip(Array(resp.data["translations"])).map do |origin, gapi|
+      def self.from_gapi_list resp, text, to, from
```
Comment (blowmage) [inline lib/gcloud/translate/translation.rb]:
``` ruby
def self.from_gapi gapi_list, text, to, from
```

**Comment by quartzmo** [comment]
@blowmage Thanks for catching these. Update coming soon.

File: lib/gcloud/translate/translation.rb (hunk 2 above)
Comment (quartzmo) [inline lib/gcloud/translate/translation.rb]: The issue here is that `Translate` has responsibility for decoding both the collection and the member. There is already a `from_gapi` for the member. What should I name the `from_gapi` for the collection? I don't really like `from_gapi_list`...

**Comment by quartzmo** [comment]
@blowmage Updated, with the Remove Errors commit squashed into the Upgrade commit.

Happy to rename `Translation.from_gapi_list` if you can suggest something. (`Translation.from_gapi` is already taken...)

**Pull request merged**